feat: stamp filtering

didierkrux opened the pull request:
I tried to implement a very simple and easy to maintain way of filtering stamps.

Feature description: https://github.com/gitcoinco/passport/issues/765

The frontend filtering seems to work well but I haven't been able to test saving the stamps because I haven't setup my `.env` file.
I'm not sure if the changes I made could break something in the saving logic and soon to be deployed stamp removal logic 🤔.

Looking forward for your feedback 🙂
Don't hesitate to contact me in the Gitcoin Passport Discord channel or DM me.
My Discord handle: `DidierKrux.eth🏴#6120`

Thanks in advance! 🙏

schultztimothy commented:
Very cool thank you for putting this together 🚀  I pulled it down and tested it locally. Everything works great!

One comment relating to how the query string is passed. We use nextjs for front end routing. Would you be willing to use a more nextjs native way of passing the query string? From what I'm seeing this seems like a possibility https://nextjs.org/docs/api-reference/data-fetching/get-initial-props

nutrina reviewed (commented):
Before we merge this I would like to have a test for this feature.
Other that that (and the comment from @schultztimothy ) looks good to me.

didierkrux commented:
I updated the PR with the following:
- rebased with the main branch + solved conflicts
- label to dynamically shows that we are only displaying filtered stamps, in our case Bankless Academy
- basic link that disables the filtering "show all my stamps"
![Screenshot 2023-02-09 at 20 20 18](https://user-images.githubusercontent.com/9531862/217935022-1e2d9503-787f-442b-a95a-38771feade62.jpg)
Let me know if you need anything else in order to merge this soon 🙏

schultztimothy commented:
👋 @didierkrux a few of the tests are failing. In order to review this on our staging environment I created a branch from your fork on the passport repo and fixed the tests https://github.com/gitcoinco/passport/tree/bankless-filtering. It would be worth pulling those changes into this branch before we merge it

didierkrux commented:
> 👋 @didierkrux a few of the tests are failing. In order to review this on our staging environment I created a branch from your fork on the passport repo and fixed the tests https://github.com/gitcoinco/passport/tree/bankless-filtering. It would be worth pulling those changes into this branch before we merge it

Thanks for your help @schultztimothy, I added your commit to this branch.
Let me know it tests work now.
